systick.c

```diff
@@ -0,0 +1,100 @@
+/*
+  Not part of Grbl. KeyMe specific.
+
+  SysTick implementation. Global SysTick value is updated.
+  Callbacks can be registed to be called after a certain time.
+
+  SysTick uses Timer1. Timer1 should not be used anywhere else.
+
+*/
+#include "systick.h"
+#include "nuts_bolts.h"
+
+#define MAX_CALLBACKS 32
+
+callback_t systick_callbacks_array[MAX_CALLBACKS];
+uint8_t systick_len;  // Length of callback array 
+
+void systick_service_callbacks()
+{
+  if (!systick_len) {
+    return;
+  }
+  // Assumes the callback array is sorted 
+  // according to lowest callback time
+  if (systick_callbacks_array[0].callback_time <= SysTick) {
```

Comment:
This should probably be a while to handle the case where two callbacks occur at the same time. The whole function could be reduced to:

```
while (systick_len && systick_callbacks_array[0].callback_time <= SysTick) {
...
}
```